### PR TITLE
Add hints about when and when not to use most/core

### PR DESCRIPTION
**Pull request by briancavalier**
We've seen folks using most/core in ways where it may not be a good fit, such as time series data processing.  I figured I'd start the discussion about how to make a short statement up front in the docs that might be helpful.

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -1,15 +1,20 @@
-.. @most/core documentation master file, created by
-   sphinx-quickstart on Mon Jun 12 07:51:42 2017.
-   You can adapt this file completely to your liking, but it should at least
-   contain the root `toctree` directive.
-
 @most/core
 ==========
 
-``@most/core`` is a toolkit for reactive programming.  It helps you write highly interactive apps by composing streams of events, and without many of the hazards of side effects and mutable shared state.
+``@most/core`` is a toolkit for reactive event programming.  It helps you write highly interactive apps by composing :ref:`event streams <concepts>`, and without many of the hazards of side effects and mutable shared state.
```
Comment (Frikki) [inline docs/index.rst]: Good one!

File: docs/index.rst [hunk 2]
```diff
@@ -1,15 +1,20 @@
...
 
+When should you use it?
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Use ``@most/core`` to build programs that transform, filter, coordinate, and *react to* :ref:`events` *as they happen*.  For example, it can be a good choice for building interactive user interfaces.
+
+When *shouldn't* you use it?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+``@most/core`` is not a good choice for use cases that aren't characterized by reacting to events as they happen.  For example, it isn't intended for Array data processing, time series data analysis and processing, or for streaming chunked data with flow control (aka "back pressure").
```
Comment (Frikki) [inline docs/index.rst]: **Wrong:** aka "“back pressure”

**Right:** a.k.a. “back pressure”

**Comment by briancavalier** [comment]
Do you think trying to represent these two sections as bulleted lists would be more helpful than as paragraphs?

File: docs/index.rst (hunk 2 above)
Comment (briancavalier) [reply docs/index.rst]: Wow, googling "aka acronym", "aka initialism", "aka vs a.k.a." yields a wide variety of info about which is "correct", and many seem to say with or without periods is equally acceptable.  There's even "a/k/a", which I'd never seen written until doing these searches!  Notably, the [Wikipedia Manual of Style](https://en.wikipedia.org/wiki/Wikipedia:Manual_of_Style/Abbreviations) suggests _never_ using "aka", and [this article says the The Chicago Manual of Style](http://blogs.millersville.edu/bduncan/abbreviations-2/) indicates using "a.k.a."

Sigh ... I decided just to spell it out 😄

File: docs/index.rst [hunk 3]
```diff
@@ -1,15 +1,20 @@
...
 
+When should you use it?
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Use ``@most/core`` to build programs that transform, filter, coordinate, and *react to* :ref:`events` *as they happen*.  For example, it can be a good choice for building interactive user interfaces.
+
+When *shouldn't* you use it?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+``@most/core`` is not a good choice for use cases that aren't characterized by reacting to events as they happen.  For example, it isn't intended for Array data processing, time series data analysis and processing, or for streaming chunked data with flow control (also known as "back pressure").
```
Comment (Frikki) [inline docs/index.rst]: I believe it is just called “Array processing” and “time series analysis and processing”.

**Comment by Frikki** [comment]
> Do you think trying to represent these two sections as bulleted lists would be more helpful than as paragraphs?

Not really. I find it hard to see how that would improve the separation between “should” and “shouldn’t”.

File: docs/index.rst (hunk 3 above)
Comment (briancavalier) [reply docs/index.rst]: Updated in 7e561ab.

**Pull request merged**